sdk/core/Azure.Core/src/Shared/OperationPoller.cs

```diff
@@ -28,10 +28,10 @@ public ValueTask<Response> WaitForCompletionResponseAsync(Operation operation, T
         public Response WaitForCompletionResponse(Operation operation, TimeSpan? suggestedInterval, CancellationToken cancellationToken)
             => WaitForCompletionAsync(false, operation, suggestedInterval, cancellationToken).EnsureCompleted();
 
-        public ValueTask<Response> WaitForCompletionResponseAsync(OperationInternalBase operation, TimeSpan? suggestedInterval, CancellationToken cancellationToken)
+        public ValueTask<Response<T>> WaitForCompletionResponseAsync<T>(OperationInternal<T> operation, TimeSpan? suggestedInterval, CancellationToken cancellationToken)
```

Comment:
Operation.WaitForCompletionResponse returns Response, where WaitForCompletion is what returns Response\<T\>.  Having a WaitForCompletionResponse\<T\> seems incorrect to me, and at a minimum can easily lead to misunderstanding of the Operation/Operation\<T\> APIs by anyone reading/maintaining this code.

Reply:
Good point. Will fix it.